conveyor:
  # Publish all joint states -----------------------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50

  # Position Controllers ---------------------------------------
  servo_front_right_position_controller:
    type: effort_controllers/JointPositionController
    joint: servo_front_right_joint
    pid: { p: 1000.0, i: 1.0, d: 0.0 }
  servo_front_left_position_controller:
    type: effort_controllers/JointPositionController
    joint: servo_front_left_joint
    pid: { p: 1000.0, i: 1.0, d: 0.0 }
  servo_back_right_position_controller:
    type: effort_controllers/JointPositionController
    joint: servo_back_right_joint
    pid: { p: 1000.0, i: 1.0, d: 0.0 }
  servo_back_left_position_controller:
    type: effort_controllers/JointPositionController
    joint: servo_back_left_joint
    pid: { p: 1000.0, i: 1.0, d: 0.0 }
  camera_position_controller:
    type: effort_controllers/JointPositionController
    joint: camera_joint
    pid: { p: 1000.0, i: 100000.0, d: 100.0 }

  # Velocity Controllers ---------------------------------------
  front_right_wheel_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: front_right_wheel_joint
    pid: {p: 1000.0, i: 1.0, d: 0.01}
  front_left_wheel_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: front_left_wheel_joint
    pid: {p: 1000.0, i: 1.0, d: 0.01}
  back_right_wheel_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: back_right_wheel_joint
    pid: {p: 1000.0, i: 1.0, d: 0.01}
  back_left_wheel_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: back_left_wheel_joint
    pid: {p: 1000.0, i: 1.0, d: 0.01}

  # gazebo_ros_control/pid_gains:
  #   front_right_wheel_joint: { p: 1.0, i: 0.0, d: 1.0 }
  #   front_left_wheel_joint: { p: 1.0, i: 0.0, d: 1.0 }
  #   back_right_wheel_joint: { p: 1.0, i: 0.0, d: 1.0 }
  #   back_left_wheel_joint: { p: 1.0, i: 0.0, d: 1.0 }

  joint_limits:
    servo_front_right_joint:
      has_position_limits: true
      # min_position: -0.785
      # max_position: 0.785
      has_velocity_limits: true
      max_velocity: 5.0
      has_acceleration_limits: true
      max_acceleration: 5.0
      has_jerk_limits: true
      max_jerk: 100.0
      has_effort_limits: true
      max_effort: 5.0
    servo_front_left_joint:
      has_position_limits: true
      # min_position: -0.785
      # max_position: 0.785
      has_velocity_limits: true
      max_velocity: 2.0
      has_acceleration_limits: true
      max_acceleration: 5.0
      has_jerk_limits: true
      max_jerk: 100.0
      has_effort_limits: true
      max_effort: 5.0
    servo_back_right_joint:
      has_position_limits: true
      # min_position: -0.785
      # max_position: 0.785
      has_velocity_limits: true
      max_velocity: 2.0
      has_acceleration_limits: true
      max_acceleration: 5.0
      has_jerk_limits: true
      max_jerk: 100.0
      has_effort_limits: true
      max_effort: 5.0
    servo_back_left_joint:
      has_position_limits: true
      # min_position: -0.785
      # max_position: 0.785
      has_velocity_limits: true
      max_velocity: 2.0
      has_acceleration_limits: true
      max_acceleration: 5.0
      has_jerk_limits: true
      max_jerk: 100.0
      has_effort_limits: true
      max_effort: 5.0

    front_right_wheel_joint:
      has_position_limits: false
      has_velocity_limits: true
      max_velocity: 50.0
      has_acceleration_limits: true
      max_acceleration: 15.0
      has_jerk_limits: true
      max_jerk: 100.0
      has_effort_limits: true
      max_effort: 5.0
    front_left_wheel_joint:
      has_position_limits: false
      has_velocity_limits: true
      max_velocity: 50.0
      has_acceleration_limits: true
      max_acceleration: 15.0
      has_jerk_limits: true
      max_jerk: 100.0
      has_effort_limits: true
      max_effort: 5.0
    back_right_wheel_joint:
      has_position_limits: false
      has_velocity_limits: true
      max_velocity: 50.0
      has_acceleration_limits: true
      max_acceleration: 15.0
      has_jerk_limits: true
      max_jerk: 100.0
      has_effort_limits: true
      max_effort: 5.0
    back_left_wheel_joint:
      has_position_limits: false
      has_velocity_limits: true
      max_velocity: 50.0
      has_acceleration_limits: true
      max_acceleration: 15.0
      has_jerk_limits: true
      max_jerk: 100.0
      has_effort_limits: true
      max_effort: 5.0
